packages/story-editor/src/components/form/tags/input.js

```diff
@@ -135,48 +187,112 @@ function Input({
           setIsInputFocused(false);
         },
       }),
-      [onUndo]
+      [onUndo, suggestedTerms]
     );
 
   const renderedTags = tagBuffer || tags;
+
+  const handleTagSelectedFromSuggestions = useCallback((e, selectedValue) => {
+    e.preventDefault();
+    setIsSuggestionsOpen(false);
+    dispatch({ type: ACTIONS.SUBMIT_VALUE, payload: selectedValue });
+    inputRef?.current.focus();
+  }, []);
+
+  const handleSuggestionKeyDown = useCallback(
+    (e, index, name) => {
+      const nextChild = index + 1;
+      const previousChild = index - 1;
+      if (e.key === 'ArrowDown') {
+        if (nextChild < totalSuggestions) {
+          menuRef?.current?.children?.[nextChild]?.focus();
+        }
+      }
+      if (e.key === 'ArrowUp') {
+        if (previousChild < 0) {
+          inputRef?.current.focus();
+        } else {
+          menuRef?.current?.children?.[previousChild]?.focus();
+        }
+      }
+      if (e.key === 'Enter') {
+        handleTagSelectedFromSuggestions(e, name);
+      }
+    },
+    [handleTagSelectedFromSuggestions, totalSuggestions]
+  );
+
   return (
-    <Border isInputFocused={isInputFocused}>
-      {
-        // Text input should move, relative to end, with offset
-        // this helps with natural tab order and visuals
-        // as you ArrowLeft or ArrowRight through tags
-        [
-          ...renderedTags.slice(0, renderedTags.length - offset),
-          INPUT_KEY,
-          ...renderedTags.slice(renderedTags.length - offset),
-        ].map((tag) =>
-          tag === INPUT_KEY ? (
-            <TextInput
-              {...props}
-              key={INPUT_KEY}
-              value={value}
-              onKeyDown={handleKeyDown}
-              onChange={handleChange}
-              onFocus={handleFocus}
-              onBlur={handleBlur}
-              size="4"
-            />
-          ) : (
-            <Tag key={tag} onDismiss={removeTag(tag)}>
-              {tagDisplayTransformer(tag) || tag}
-            </Tag>
+    <Border ref={containerRef} isInputFocused={isInputFocused}>
+      <InputWrapper>
+        {
+          // Text input should move, relative to end, with offset
+          // this helps with natural tab order and visuals
+          // as you ArrowLeft or ArrowRight through tags
+          [
+            ...renderedTags.slice(0, renderedTags.length - offset),
+            INPUT_KEY,
+            ...renderedTags.slice(renderedTags.length - offset),
+          ].map((tag) =>
+            tag === INPUT_KEY ? (
+              <TextInput
+                {...props}
+                key={INPUT_KEY}
+                value={value}
+                onKeyDown={handleKeyDown}
+                onChange={handleChange}
+                onFocus={handleFocus}
+                onBlur={handleBlur}
+                size="4"
+                ref={inputRef}
+                autoComplete={isSuggestionsOpen ? 'off' : 'on'}
```

Comment:
💭 After playing around with this I think we also need the `aria-expanded` flag to get the screen reader to read it as open/close
```suggestion
                autoComplete={isSuggestionsOpen ? 'off' : 'on'}
                aria-expanded={isSuggestionsOpen}
```

Reply:
you're right. So i'd done this and then I ran aXe and got yelled at for it. Dug into it briefly today since tomorrow I'm stuck at jury duty and it's because the input role needed to be updated to a combobox. Thanks for keeping this in mind. should be fixed!